yamllint/config.py

```diff
@@ -97,12 +97,30 @@ def parse(self, raw_content):
             except Exception as e:
                 raise YamlLintConfigError('invalid config: %s' % e)
 
-        if 'ignore' in conf:
-            if not isinstance(conf['ignore'], str):
-                raise YamlLintConfigError(
-                    'invalid config: ignore should contain file patterns')
-            self.ignore = pathspec.PathSpec.from_lines(
-                'gitwildmatch', conf['ignore'].splitlines())
+        if ('ignore' in conf) ^ ('ignore-from-file' in conf):
+
+            if 'ignore-from-file' in conf:
+                if isinstance(conf['ignore-from-file'], str):
+                    # if str enclose in list
+                    conf['ignore-from-file'] = [conf['ignore-from-file']]
+                self.ignore = pathspec.PathSpec.from_lines(
+                    'gitwildmatch',
+                    _concat_files_content(
+                        conf['ignore-from-file']
+                    ).splitlines()
+                )
+
+            elif 'ignore' in conf:
+                if not isinstance(conf['ignore'], str):
+                    raise YamlLintConfigError(
+                        'invalid config: ignore should contain file patterns')
+                self.ignore = pathspec.PathSpec.from_lines(
+                    'gitwildmatch', conf['ignore'].splitlines())
+        elif 'ignore' in conf and 'ignore-from-file' in conf:
+            raise YamlLintConfigError(
+                'invalid config: '
+                'ignore and ignore-from-file keys cannot be used together'
+            )
```

Comment:
This form is shorter, maybe it's also more readable:
```python
        if 'ignore' in conf and 'ignore-from-file' in conf:
	    raise YamlLintConfigError(...)
        elif 'ignore-from-file' in conf:
	    pass
        elif 'ignore' in conf:
	    pass
```

Reply:
> Before review starts, can you rebase on latest `master`, so that tests are run? We just switched today from Travis to GitHub Actions. Also you can use a conventional commit title for easier processing (e.g. `git blame`/`log`/`bisect`...), see examples in this repo commit history; and please add the full URL `https://github.com/adrienverge/yamllint/issues/360` in the commit message.

You're welcome!

Sure, fixed that. 



>  I don't know what to think about accepting both _filenames_ and _lists of filenames_.

I wasn't sure too, just wanted to keep balance between user-friendliness and precise types. However, I put the string into list in parsing. This will help in removing it in case the disadvantages weighs the gains. 



> Have you tested all possible corner cases? E.g. incorrect files passed to `ignore-from-file`? Multiple `ignore-from-file` that override themselves? Etc.

Yes, for incorrect files passed to `ignore-from-file`. But, for duplicate keys I leave it on `yaml` since you use it, which will use only the latest key in the file. If you expect something else, please let me know so that I refactor the code to conform with it.